RoddyCore/src/de/dkfz/roddy/knowledge/methods/GenericMethod.groovy

```diff
@@ -334,24 +368,41 @@ class GenericMethod {
         return fileParameter.fileClass.cast(bf) as FileObject;
     }
 
-    private FileObject createOutputTuple(ToolEntry.ToolTupleParameter tfg) {
+    FileObject createOutputTuple(ToolTupleParameter tfg) {
         //TODO Auto recognize tuples?
         List<FileObject> filesInTuple = [];
-        for (ToolEntry.ToolFileParameter fileParameter in tfg.files) {
+        for (ToolFileParameter fileParameter in tfg.files) {
             BaseFile bf = convertToolFileParameterToBaseFile(fileParameter);
             filesInTuple << bf;
             allCreatedObjects << bf;
         }
         return FileObjectTupleFactory.createTuple(filesInTuple);
     }
 
-    private FileObject createOutputFileGroup(ToolEntry.ToolFileGroupParameter tfg) {
+    FileObject createOutputFileGroup(ToolFileGroupParameter tfg) {
         List<BaseFile> filesInGroup = [];
 
-        for (ToolEntry.ToolFileParameter fileParameter in tfg.files) {
-            BaseFile bf = convertToolFileParameterToBaseFile(fileParameter)
-            filesInGroup << bf;
-            allCreatedObjects << bf;
+        if (tfg.files) {
+            // Actually this is more like a tuple.
+            /**
+             * This can only be the case if files is set. Otherwise we need a different way to identify files. E.g. based on index values... How do we set them?
+             */
+            for (ToolFileParameter fileParameter in tfg.files) {
```

Comment:
Actually I took that out yesterday.?? Don't know why it is still in here...

`private FileObject createOutputFileGroup(ToolEntry.ToolFileGroupParameter tfg) {
        List<BaseFile> filesInGroup = [];

        for (ToolEntry.ToolFileParameter fileParameter in tfg.files) {
            BaseFile bf = convertToolFileParameterToBaseFile(fileParameter)
            filesInGroup << bf;
            allCreatedObjects << bf;
        }

        Constructor cGroup = tfg.groupClass.getConstructor(List.class);
        FileGroup bf = cGroup.newInstance(filesInGroup);
        return tfg.groupClass.cast(bf) as FileObject;
    }
`